extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts

```diff
@@ -0,0 +1,328 @@
+    }
+
+    removeRegistry(registry: ChePluginRegistry): void {
+        this.registryList = this.registryList.filter(r => r.uri !== registry.uri);
+    }
+
+    getRegistryList(): ChePluginRegistry[] {
+        return this.registryList;
+    }
+
+    /**
+     * Returns plugin list from active registry
+     */
+    async getPlugins(filter: string): Promise<ChePluginMetadata[]> {
+        await this.initDefaults();
+
+        this.availablePlugins = await this.chePluginService.getPlugins(this.activeRegistry, filter);
+        // The code will be use soon
+        // const deployedPlugins = await this.pluginFrontentService.getDeployedPlugins(filter);
+        // this.availablePlugins = this.availablePlugins.concat(deployedPlugins);
+        return this.availablePlugins;
+    }
+
+    isPluginInstalled(plugin: ChePluginMetadata): boolean {
+        return this.installedPlugins.indexOf(plugin.key) >= 0;
+    }
+
+    async install(plugin: ChePluginMetadata): Promise<boolean> {
+        try {
+            // add the plugin to workspace configuration
+            await this.chePluginService.addPlugin(plugin.key);
+            await this.delay(1000);
+            this.messageService.info(`Plugin '${plugin.publisher}/${plugin.name}/${plugin.version}' has been successfully installed`);
+
+            // add the plugin to the list of workspace plugins
+            this.installedPlugins.push(plugin.key);
+
+            // notify that workspace configuration has been changed
+            this.notifyWorkspaceConfigurationChanged();
+            return true;
+        } catch (error) {
+            this.messageService.error(`Unable to install plugin '${plugin.publisher}/${plugin.name}/${plugin.version}'. ${error.message}`);
+            return false;
+        }
+    }
+
+    async remove(plugin: ChePluginMetadata): Promise<boolean> {
+        try {
+            // remove the plugin from workspace configuration
+            await this.chePluginService.removePlugin(plugin.key);
+            await this.delay(1000);
+            this.messageService.info(`Plugin '${plugin.publisher}/${plugin.name}/${plugin.version}' has been successfully removed`);
+
+            // remove the plugin from the list of workspace plugins
+            this.installedPlugins = this.installedPlugins.filter(p => p !== plugin.key);
+
+            // notify that workspace configuration has been changed
+            this.notifyWorkspaceConfigurationChanged();
+            return true;
+        } catch (error) {
+            this.messageService.error(`Unable to remove plugin '${plugin.publisher}/${plugin.name}/${plugin.version}'. ${error.message}`);
+            return false;
+        }
+    }
+
+    private getIdPublisher(input: string): string {
+        if (input.startsWith('ext install ')) {
+            // check for 'ext install rebornix.Ruby'
+            return input.substring('ext install '.length);
```

Comment:
It would be nice to trim result.

Reply:
This is intentional decision. Space is a part of a prefix of the command.
The first command has a space at the end, the second command has a slash.
```
        if (input.startsWith('ext install ')) {
            // check for 'ext install rebornix.Ruby'
            return input.substring('ext install '.length);
        } else if (input.startsWith('vscode:extension/')) {
            // check for 'vscode:extension/rebornix.Ruby'
            return input.substring('vscode:extension/'.length);
        }
```